test/specialfunctionstests.jl

```diff
@@ -0,0 +1,19 @@
+using FastTransforms
+using Compat.Test
+import FastTransforms.pochhammer
+
+const sqrtpi = 1.772453850905516027298
+
+@testset "Pochhammer" begin
+    @test pochhammer(2,3) == 24
+    @test pochhammer(0.5,3) == 0.5*1.5*2.5
+    @test pochhammer(0.5,0.5) == 1/sqrtpi
+    @test pochhammer(0,1) == 0
+    @test pochhammer(-1,2) == 0
+    @test pochhammer(-5,3) == -60
+    @test pochhammer(-1,-0.5) == 0
+    @test 1.0/pochhammer(-0.5,-0.5) == 0
+    @test pochhammer(-1+0im,-1) == -2
+    @test pochhammer([2,-5,-1],3) == [24,-60,0]
```

Comment:
vector-valued arguments should be deprecated in favour of `pochhammer.([2,-5,-1],3)`, so perhaps remove this test.

Reply:
So can `pochhammer(x::AbstractArray{T},n::Integer)` be removed? Do abstractarrays work with dot operations?

Reply:
It can be deprecated, though that can be done separately. Yes abstract arrays support broadcastibg